src/pages/signup-affiliates/components/account-details.tsx

```diff
@@ -0,0 +1,349 @@
+            type: 'text',
+            label: is_individual
+                ? localize('_t_Second website/social media URL_t_')
+                : localize('_t_Company second website/social media URL_t_'),
+            error: second_website_url_error_msg,
+            value: second_website_url,
+            required: false,
+            value_set: setSecondWebsiteUrl,
+            error_set: setSecondWebsiteUrlErrorMsg,
+        },
+        {
+            id: 'username',
+            name: 'username',
+            type: 'text',
+            label: localize('_t_Username*_t_'),
+            error: user_name_error_msg,
+            value: username,
+            required: true,
+            value_set: setUsername,
+            error_set: setUserNameErrorMsg,
+        },
+        {
+            id: 'dm-password',
+            name: 'password',
+            type: 'password',
+            label: localize('_t_Password*_t_'),
+            error: password_error_msg,
+            value: password,
+            required: true,
+            value_set: setPassword,
+            error_set: setPasswordErrorMsg,
+        },
+    ]
+    // prevent these characters from the input type='number' tag
+    const input_id = document.getElementById('company_registration_number')
+    const phone_id = document.getElementById('phone')
+    input_id?.addEventListener('keydown', function (e) {
+        if (['.', ',', '-', '+', 'e'].includes(e.key)) {
+            e.preventDefault()
+        }
+    })
+    phone_id?.addEventListener('keydown', function (e) {
+        if (['.', ',', '-', '+', 'e'].includes(e.key)) {
+            e.preventDefault()
+        }
+    })
+
+    const validate = is_individual
+        ? username &&
+          first_name &&
+          last_name &&
+          date_birth &&
+          phone &&
+          password &&
+          !first_name_error_msg &&
+          !last_name_error_msg &&
+          !password_error_msg
+        : first_name &&
+          last_name &&
+          date_birth &&
+          password &&
+          company_name &&
+          company_registration_number &&
+          !first_name_error_msg &&
+          !last_name_error_msg &&
+          !phone_error_msg &&
+          !password_error_msg &&
+          !company_name_error_msg &&
+          !company_registration_error_msg
+
+    useEffect(() => {
+        updateData({
+            ...affiliate_data,
+            first_name,
+            last_name,
+            date_birth,
+            phone,
+            prefix,
+            company_name,
+            company_registration_number,
+            website_url,
+            second_website_url,
+            username,
+            password,
+        })
+    }, [
+        username,
+        first_name,
+        last_name,
+        date_birth,
+        phone,
+        company_name,
+        company_registration_number,
+        website_url,
+        second_website_url,
+        password,
+    ])
+
+    useEffect(() => {
```

Comment:
In general I think we should only `useEffect` when touching something that's outside of React's control.

Would it be possible to call `onValidate` directly within the handlers where the validation-relevant state changes occur?

Instead of using useEffect, you can directly call onValidate within the state update functions where the validation state changes. For example, in the handleInput function, after setting the state (like setFirstName(value)), you can also call onValidate with the updated validation state. This approach ensures that validation occurs precisely when relevant data changes, avoiding the blanket catch-all nature of useEffect.